src/backend/joanie/tests/core/test_api_enrollment.py

```diff
@@ -1226,3 +1261,172 @@ def test_api_enrollment_update_detail_authenticated_owner(self, _mock_set):
+                    ),
+                    "start": course_run.start.isoformat().replace("+00:00", "Z"),
+                    "end": course_run.end.isoformat().replace("+00:00", "Z"),
+                    "state": {
+                        "priority": course_run.state["priority"],
+                        "text": course_run.state["text"],
+                        "call_to_action": course_run.state["call_to_action"],
+                        "datetime": course_run.state["datetime"]
+                        .isoformat()
+                        .replace("+00:00", "Z"),
+                    },
+                },
+                "is_active": False,
+                "state": "set",
+                "was_created_by_order": False,
+            },
+        )
+
+        # Then user purchases a product containing the previously created course run and
+        # tries to update to was_created_by_order field again.
+        factories.CourseRunFactory(course=course_run.course)
+        product = factories.ProductFactory(target_courses=[course_run.course])
+        order = factories.OrderFactory(owner=user, product=product)
+        # - Create an invoice related to the order to mark it as validated
+        InvoiceFactory(order=order, total=order.total)
+
+        response = self.client.put(
+            f"/api/v1.0/enrollments/{enrollment.id}/",
+            data={
+                "is_active": True,
+                "was_created_by_order": True,
+            },
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {user_token}",
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.json(),
+            {
+                "id": str(enrollment.id),
+                "course_run": {
+                    "id": str(course_run.id),
+                    "resource_link": course_run.resource_link,
+                    "title": course_run.title,
+                    "enrollment_start": course_run.enrollment_start.isoformat().replace(
+                        "+00:00", "Z"
+                    ),
+                    "enrollment_end": course_run.enrollment_end.isoformat().replace(
+                        "+00:00", "Z"
+                    ),
+                    "start": course_run.start.isoformat().replace("+00:00", "Z"),
+                    "end": course_run.end.isoformat().replace("+00:00", "Z"),
+                    "state": {
+                        "priority": course_run.state["priority"],
+                        "text": course_run.state["text"],
+                        "call_to_action": course_run.state["call_to_action"],
+                        "datetime": course_run.state["datetime"]
+                        .isoformat()
+                        .replace("+00:00", "Z"),
+                    },
+                },
+                "is_active": True,
+                "state": "set",
+                "was_created_by_order": True,
+            },
+        )
```

Comment:
Just to have all the cases, I would add the test that tries to update the field was_created_by_order to False to show it's not possible as well if this enrollment is active for enrollment that is in the scope of an order. This way we will have the update tested for free Enrollment and for order.